pkg/analyser/analyzer.go

```diff
@@ -42,18 +42,17 @@ type AnalyzerOptions struct {
 }
 
 type Analyzer struct {
-	alerter *alerter.Alerter
-	options AnalyzerOptions
-	filter  filter.Filter
+	analysis *Analysis
```

Comment:
not sure why we need an analysis in here and method return pointer to potentially incomplete analysis

Reply:
Because we need to keep the same analysis between CompareEnumeration and CompareDetails

Reply:
we could take the old analysis in arg, if you want to keep a local analysis inside the object you should not return a pointer of it to the outside this is error prone and that's also what led to having an api like
```go
analysis := anaylzer.CompareEnumeration()
// analysis does not have correct alert a this point
anaylzer.SetAlerts()
// Now we didn't do anything to analysis from our pov but the alert are set
```